flake.nix

```diff
@@ -0,0 +1,107 @@
+{
+  description = "A simple terminal UI for git commands";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-25.05";
+    systems.url = "github:nix-systems/default";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-compat.url = "https://flakehub.com/f/edolstra/flake-compat/1.tar.gz";
+    treefmt-nix.url = "github:numtide/treefmt-nix";
+  };
+
+  outputs =
+    inputs@{ flake-parts, systems, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = import systems;
+      imports = [
+        inputs.treefmt-nix.flakeModule
+      ];
+
+      perSystem =
+        {
+          pkgs,
+          ...
+        }:
+        let
+          lazygit = pkgs.buildGoModule rec {
+            pname = "lazygit";
+            version = "dev";
+
+            gitCommit = inputs.self.rev or inputs.self.dirtyRev or "dev";
+
+            src = ./.;
+            vendorHash = null;
+
+            # Disable integration tests that require specific environment
+            doCheck = false;
+
+            nativeBuildInputs = with pkgs; [
+              git
+              makeWrapper
+            ];
+            buildInputs = [ pkgs.git ];
+
+            ldflags = [
+              "-s"
+              "-w"
+              "-X main.commit=${gitCommit}"
+              "-X main.version=${version}"
+              "-X main.buildSource=nix"
+            ];
+
+            postInstall = ''
+              wrapProgram $out/bin/lazygit \
+                --prefix PATH : ${pkgs.lib.makeBinPath [ pkgs.git ]}
+            '';
+
+            meta = {
+              description = "A simple terminal UI for git commands";
+              homepage = "https://github.com/jesseduffield/lazygit";
+              license = pkgs.lib.licenses.mit;
+              maintainers = [ "jesseduffield" ];
+              platforms = pkgs.lib.platforms.unix;
+              mainProgram = "lazygit";
+            };
+          };
+        in
+        {
+          packages = {
+            default = lazygit;
+            inherit lazygit;
+          };
+
+          devShells.default = pkgs.mkShell {
+            name = "lazygit-dev";
+
+            buildInputs = with pkgs; [
+              # Go toolchain
+              go
+              gotools
+              golangci-lint
```

Comment:
For golangci-lint it is even more important than for go itself to be on the right version. Which checks are enabled by default tends to change between versions, so you are likely to get spurious linter warnings when using the latest version.

We already have a mechanism in place to mitigate this by using a shim that runs golangci-lint with the right version (see `make lint` in the Makefile), so I wonder if we should remove this entirely rather than add an explicit version.

Reply:
Got it. Yes, then we can remove this from the nix file as I'm including gnumake. Will wait for unresolved comments to be updated before pushing a chore commit and rebasing.